python/composio/tools/base/abs.py

```diff
@@ -40,6 +40,7 @@ def remove_json_ref(data: t.Dict) -> t.Dict:
             jsonref.replace_refs(
                 obj=data,
                 lazy_load=False,
+                merge_props=True,
```

Comment:
Consider adding a comment explaining why `merge_props=True` is necessary here. This parameter is crucial for handling nested schemas correctly, and documenting this requirement would help future maintainers understand the importance of this setting.